app/models/types/code.rb

```diff
@@ -0,0 +1,46 @@
+class Code
+
+  attr_reader :code, :code_system, :descriptor, :code_system_oid, :version
+
+  # Code and code system are required (at minimum).
+  def initialize(code, code_system, descriptor=nil, code_system_oid=nil, version=nil)
+    @code = code
+    @code_system = code_system
+    @descriptor = descriptor
+    @code_system_oid = code_system_oid
+    @version = version
+  end
+
+  # Converts an object of this instance into a database friendly value.
+  def mongoize
+    [ code, code_system, descriptor, code_system_oid, version ]
+  end
+
+  class << self
+
+    # Get the object as it was stored in the database, and instantiate
+    # this custom class from it.
+    def demongoize(object)
+      Code.new(object[0], object[1], object[2], object[3], object[4])
```

Comment:
Do you want to add something to the comment for this that says "The array indices in `demongoize` are the same 5 elements used in `mongoize`" or the like?

Reply:
Yes, I think that clarification make sense. Added.